src/components/loan-investment-calculator/index.js

```diff
@@ -0,0 +1,220 @@
+        option1.pmt,
+        0
+      )
+    }
+  }
+
+  // Returns opportunity cost of choosing option1 over option2. Opportunity cost can be positive and negative.
+  calculateOpportunityCost (option1, option2) {
+    return Math.abs(option1.fv - option2.fv)
+  }
+
+  // Runs every time an input is updated and uses input's name tag to make specific changes in the state
+  updateInput (e) {
+    let state = Object.assign({}, this.state)
+    const key = e.target.name
+
+    const value = parseFloat(e.target.value)
+
+    _.set(state, key, value)
+    if (!isNaN(value)) {
+      state = this.calculateAll(state)
+    }
+    this.setState(state)
+  }
+
+  // Runs all calculations and returns a modified state
+  calculateAll (state) {
+    option1 = state.options.option1
+    option2 = state.options.option2
+
+    option1.pmt = this.calculatePMT(option1, state)
+    option2.pmt = this.calculatePMT(option2, state)
+
+    option1.interestAmt = this.calculateInterestAmt(option1, state)
+    option2.interestAmt = this.calculateInterestAmt(option2, state)
+
+    option1.fv = this.calculateFV(option1, option2, state)
+    option2.fv = this.calculateFV(option2, option1, state)
+
+    state.optCost = this.calculateOpportunityCost(option1, option2)
+
+    const [shorter, longer] = _.sortBy([option1, option2], 'term')
+    state.shorterOption = shorter.term === option1.term ? 'option1' : 'option2'
+    state.longerOption = longer.term === option1.term ? 'option1' : 'option2'
+
+    state.yearlyResultsByOption = getYearly({
+      loanAmt: state.loanAmt,
+      investmentRate: state.investmentRate / 100,
+      inflationRate: state.inflation / 100,
+      shorterOption: {
+        mortgageRate: shorter.interestRate / 100,
+        mortgageTerm: shorter.term
+      },
+      longerOption: {
+        mortgageRate: longer.interestRate / 100,
+        mortgageTerm: longer.term
+      }
+    })
+
+    return state
+  }
+
+  render () {
+    const shorterOption = this.state.options[this.state.shorterOption]
+    const longerOption = this.state.options[this.state.longerOption]
+    const bestOption = _.maxBy([shorterOption, longerOption], 'fv')
+
+    return (
+      <Row>
+        <Col xs='4'>
+          <InputCard
+            loanAmt={this.state.loanAmt}
+            inflation={this.state.inflation}
+            investmentRate={this.state.investmentRate}
+            updateInput={this.updateInput}
+            option1={option1}
+            option2={option2}
+            shorterOption={shorterOption}
+            longerOption={longerOption}
+          />
+        </Col>
+        <Col xs='8'>
+          <Summary
+            shorterOption={shorterOption}
+            longerOption={longerOption}
+            bestOption={bestOption}
+            optCost={this.state.optCost}
+            inflation={this.state.inflation}
+            investmentRate={this.state.investmentRate}
+            yearlyResultsByOption={this.state.yearlyResultsByOption}
+          />
```

Comment:
The only issue w/ `loan-investment-calculator` is that the summary has copy (and maybe the input card too) that is specific to mortgages. How about `mortgage-investment-compare`?

Reply:
Yes, you are right. Could we make it `loan-investment-compare` instead? The calculator itself can be used with anything and changing the text in the future is easy. I'd like to keep it general to remind us that we can generalize the tool later on for any other types of loans (e.g. question about comparing car loans in personal finance forum). We can frame it for mortgages, but it's still a loan-investment comparison tool if that makes sense.

Reply:
> I'd like to keep it general to remind us that we can generalize the tool later on for any other types of loans (e.g. question about comparing car loans in personal finance forum).

I get the intent, though that may be a bit more abstract than needed at this point. I don't think we'll forget to make more abstractions if/when we get to that, and currently the name can be misleading to a reader since it isn't built out generically